src/main/java/org/commcare/formplayer/installers/FormplayerLocaleInstaller.java

```diff
@@ -0,0 +1,102 @@
+package org.commcare.formplayer.installers;
+
+import org.commcare.resources.ResourceInstallContext;
+import org.commcare.resources.model.*;
+import org.commcare.util.CommCarePlatform;
+import org.javarosa.core.reference.InvalidReferenceException;
+import org.javarosa.core.reference.Reference;
+import org.javarosa.core.services.locale.Localization;
+import org.javarosa.core.services.locale.LocalizationUtils;
+import org.javarosa.core.services.locale.TableLocaleSource;
+import org.javarosa.core.util.externalizable.DeserializationException;
+import org.javarosa.core.util.externalizable.ExtUtil;
+import org.javarosa.core.util.externalizable.ExtWrapMap;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+import org.javarosa.xml.util.InvalidStructureException;
+import org.javarosa.xml.util.UnfullfilledRequirementsException;
+import org.xmlpull.v1.XmlPullParserException;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Hashtable;
+import java.util.Vector;
+
+/**
+ * ResourceInstaller for locale files that always persists the locale data in
+ * the SQLite DB.
+ */
+public class FormplayerLocaleInstaller extends SimpleInstaller {
+
+    private String locale;
+
+    private Hashtable<String, String> localizedValues;
+
+    /**
+     * Serialization only!
+     */
+    @SuppressWarnings("unused")
+    public FormplayerLocaleInstaller() {
+    }
+
+    public FormplayerLocaleInstaller(String locale) {
+        this.locale = locale;
+    }
+
+    @Override
+    public boolean initialize(CommCarePlatform platform, boolean isUpgrade) throws IOException, InvalidReferenceException, InvalidStructureException, XmlPullParserException, UnfullfilledRequirementsException {
+        Localization.getGlobalLocalizerAdvanced().addAvailableLocale(locale);
+        Localization.getGlobalLocalizerAdvanced().registerLocaleResource(locale, new TableLocaleSource(localizedValues));
```

Comment:
probably need to check if this is null and if not pass in an empty hashmap

Reply:
Since the installer is committed atomically with the resource record, I don't think there's a circumstance where the table could end up null after installed as long as `parseLocaleInput` passes back a valid table. Probably best to push any validation into the installer and fail fast if anything is off.